core/src/main/java/nl/nn/adapterframework/http/rest/ApiServiceDispatcher.java

```diff
@@ -186,33 +217,32 @@ protected JsonObject generateOpenApiJsonSchema(Collection<ApiDispatchConfig> cli
 		JsonObjectBuilder paths = Json.createObjectBuilder();
 		JsonObjectBuilder schemas = Json.createObjectBuilder();
 
-		for (ApiDispatchConfig config : clients) {
+		for (Entry<String, List<ApiListener>> config : groupedByUri.entrySet()) {
 			JsonObjectBuilder methods = Json.createObjectBuilder();
-			ApiListener listener = null;
-			for (HttpMethod method : config.getMethods()) {
+			for (ApiListener listener : config.getValue()) {
 				JsonObjectBuilder methodBuilder = Json.createObjectBuilder();
-				listener = config.getApiListener(method);
-				if(listener != null && listener.getReceiver() != null) {
-					IAdapter adapter = listener.getReceiver().getAdapter();
-					if (StringUtils.isNotEmpty(adapter.getDescription())) {
-						methodBuilder.add("summary", adapter.getDescription());
-					}
-					if(StringUtils.isNotEmpty(listener.getOperationId())) {
-						methodBuilder.add("operationId", listener.getOperationId());
-					}
-					// GET and DELETE methods cannot have a requestBody according to the specs.
-					if(method != HttpMethod.GET && method != HttpMethod.DELETE) {
-						mapRequest(adapter, listener.getConsumes(), methodBuilder);
-					}
-					mapParamsInRequest(adapter, listener, methodBuilder);
 
-					methodBuilder.add("responses", mapResponses(adapter, listener.getContentType(), schemas));
+				if(listener.getReceiver() == null) continue;
```

Comment:
I'd suggest first filtering, then creating the method-builder instance.

Also, I think this loop is a good candidate for converting to Streaming API, using a combination of filter/map/forEach.